src/v/cluster/types.h

```diff
@@ -1312,6 +1315,31 @@ struct topic_configuration
         return std::tie(tp_ns, partition_count, replication_factor, properties);
     }
 
+    void serde_read(iobuf_parser& in, const serde::header& h) {
```

Comment:
this serde_read does less checking than it would be done without it, not sure if it matters 
https://github.com/redpanda-data/redpanda/blob/b0facb575cc8112ef2898f2212ad7dd7a8f72d7d/src/v/serde/serde.h#L653